### PR TITLE
Add 'unused' lint

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: src/github.com/matrix-org/dendrite/cmd/syncserver-integration-tests/testdata.go [hunk 1]
```diff
@@ -35,15 +35,6 @@ const (
...
 	i18StateAliceRoomName
-	i19BobMsg
-	i20StateBobLeave
-	i21AliceMsg
-	i22StateAliceInviteBob
-	i23StateBobRejectInvite
-	i24AliceMsg
-	i25StateAliceRoomName
-	i26StateCharlieJoin
-	i27CharlieMsg
```
Comment (NegativeMjark) [inline src/github.com/matrix-org/dendrite/cmd/syncserver-integration-tests/testdata.go]: Can we keep these somehow? I suspect they might be the only nice way to identify the test data here.

**Comment by richvdh** [comment]
ok to test

File: src/github.com/matrix-org/dendrite/cmd/syncserver-integration-tests/testdata.go (hunk 1 above)
Comment (erikjohnston) [reply src/github.com/matrix-org/dendrite/cmd/syncserver-integration-tests/testdata.go]: I've kept them, though I find the entire file deeply confusing.

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**